titus-ext/jooq/src/main/java/com/netflix/titus/ext/jooq/relocation/JooqTaskRelocationGC.java

```diff
@@ -0,0 +1,167 @@
+    private final JooqTaskRelocationResultStore relocationResultStore;
+    private final TitusRuntime titusRuntime;
+
+    private ScheduleReference scheduleRef;
+
+    private final Gauge allRowsGauge;
+    private final Gauge expiredRowsGauge;
+    private final Counter gcCounter;
+
+    @Inject
+    public JooqTaskRelocationGC(JooqRelocationConfiguration configuration,
+                                DSLContext dslContext,
+                                JooqTaskRelocationResultStore relocationResultStore,
+                                TitusRuntime titusRuntime) {
+        this.configuration = configuration;
+        this.dslContext = dslContext;
+        this.relocationResultStore = relocationResultStore;
+        this.titusRuntime = titusRuntime;
+        this.allRowsGauge = titusRuntime.getRegistry().gauge(METRIC_ROOT + "allRows", "table", "relocation_status");
+        this.expiredRowsGauge = titusRuntime.getRegistry().gauge(METRIC_ROOT + "expiredRows", "table", "relocation_status");
+        this.gcCounter = titusRuntime.getRegistry().counter(METRIC_ROOT + "removedCount", "table", "relocation_status");
+    }
+
+    @Override
+    public void activate() {
+        this.scheduleRef = titusRuntime.getLocalScheduler().schedule(ScheduleDescriptor.newBuilder()
+                        .withName(JooqTaskRelocationGC.class.getSimpleName())
+                        .withDescription("Titus relocation data GC")
+                        .withInitialDelay(Duration.ZERO)
+                        .withInterval(configuration.getGcInterval())
+                        .withTimeout(Duration.ofMinutes(30))
+                        .build(),
+                context -> doGC(),
+                true
+        );
+    }
+
+    @PreDestroy
+    public void shutdown() {
+        Evaluators.acceptNotNull(scheduleRef, ScheduleReference::cancel);
+    }
+
+    private void doGC() {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        logger.info("Starting new relocation status table GC cycle...");
+        removeExpiredData(titusRuntime.getClock().wallTime() - configuration.getRetentionTime().toMillis());
+        logger.info("The relocation status table GC cycle finished: elapsedMs={}", stopwatch.elapsed(TimeUnit.MILLISECONDS));
+    }
+
+    /**
+     * Removes all entries older than the given time threshold.
+     */
+    @VisibleForTesting
+    int removeExpiredData(long timeThreshold) {
+        // Count all items
+        int allCount = dslContext.fetchCount(RELOCATION_STATUS);
+        logger.info("All rows in 'relocation_status' table: {}", allCount);
+        allRowsGauge.set(allCount);
+
+        int expiredCount = dslContext.fetchCount(
+                RELOCATION_STATUS,
+                RELOCATION_STATUS.RELOCATION_EXECUTION_TIME.lt(Timestamp.from(Instant.ofEpochMilli(timeThreshold)))
+        );
+        logger.info("Expired rows in 'relocation_status' table: {}", expiredCount);
+        expiredRowsGauge.set(allCount);
```

Comment:
copy-and-pasta error? I belive the argument here should be `expiredCount`: `expiredRowsGauge.set(expiredCount)`